src/stdlib/parse_cbor.rs

```diff
@@ -0,0 +1,125 @@
+use crate::compiler::prelude::*;
+use ciborium::de::from_reader;
+use zstd::zstd_safe::WriteBuf;
+
+fn parse_cbor(value: Value) -> Resolved {
+    let bytes = value.try_bytes()?;
+    let value = from_reader(bytes.as_slice()).map_err(|e| format!("unable to parse cbor: {e}"))?;
+    Ok(value)
+}
+
+#[derive(Clone, Copy, Debug)]
+pub struct ParseCbor;
+
+impl Function for ParseCbor {
+    fn identifier(&self) -> &'static str {
+        "parse_cbor"
+    }
+
+    fn summary(&self) -> &'static str {
+        "parse a string to a CBOR type"
+    }
+
+    fn usage(&self) -> &'static str {
+        indoc! {"
+            Parses the provided `value` as CBOR.
+        "}
+    }
+
+    fn examples(&self) -> &'static [Example] {
+        &[Example {
+            title: "object",
+            source: r#"parse_cbor!(decode_base64!("oWVmaWVsZGV2YWx1ZQ=="))"#,
+            result: Ok(r#"{ "field": "value" }"#),
+        }]
+    }
+
+    fn compile(
+        &self,
+        _state: &state::TypeState,
+        _ctx: &mut FunctionCompileContext,
+        arguments: ArgumentList,
+    ) -> Compiled {
+        let value = arguments.required("value");
+        Ok(ParseCborFn { value }.as_expr())
+    }
+
+    fn parameters(&self) -> &'static [Parameter] {
+        &[Parameter {
+            keyword: "value",
+            kind: kind::BYTES,
+            required: true,
+        }]
+    }
+}
+
+#[derive(Debug, Clone)]
+struct ParseCborFn {
+    value: Box<dyn Expression>,
+}
+
+impl FunctionExpression for ParseCborFn {
+    fn resolve(&self, ctx: &mut Context) -> Resolved {
+        let value = self.value.resolve(ctx)?;
+        parse_cbor(value)
+    }
+
+    fn type_def(&self, _: &state::TypeState) -> TypeDef {
+        type_def()
+    }
+}
+
+fn inner_kind() -> Kind {
+    Kind::null()
+        | Kind::bytes()
+        | Kind::integer()
+        | Kind::float()
+        | Kind::boolean()
+        | Kind::array(Collection::any())
+        | Kind::object(Collection::any())
+}
+
+fn type_def() -> TypeDef {
```

Comment:
Is the intention for this and `inner_kind()` to be identical to JSON type defs?

Reply:
Yes, cbor uses the same data model as JSON (https://cbor.io/)
I will provide more tests to cover other types besides object and create a PR with documentation to the main vector repository

Reply:
Thank you for the explanation. 

TIOLI: we can extract these type defs under a new directory `src/stdlib/json_utils`.

Reply:
Moved type defs to separate module and provided more examples for parse_cbor